apps/nextjs/src/components/AppComponents/Chat/chat-lessonPlanDisplay.tsx

```diff
@@ -109,7 +144,24 @@ export const LessonPlanDisplay = ({
     documentContainerRef,
   ]);
 
+  const sectionsToDisplay = getSectionsToDisplay(lessonPlan, streamingSections);
+
+  const [prevSectionsToDisplay, setPrevSectionsToDisplay] = useState<string[]>(
+    [],
+  );
+
+  useEffect(() => {
+    if (
+      JSON.stringify(sectionsToDisplay) !==
+      JSON.stringify(prevSectionsToDisplay)
+    ) {
+      console.log("Render LessonPlanDisplay with sections", sectionsToDisplay);
+      setPrevSectionsToDisplay(sectionsToDisplay);
+    }
+  }, [sectionsToDisplay, prevSectionsToDisplay]);
```

Comment:
If planning to keep this, it would be worth putting it in a hook in this file. It doesn't look like we need to use `prevSectionsToDisplay` in the render itself

I think `prevSectionsToDisplay` would be better as a ref, to avoid a second render

It's maybe a bit misleading to log "Render", as there are other reasons it would render. It's more that the sections have changed